libs/ledger/include/ledger/chain/time_travelogue.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "chain/constants.hpp"
+#include "core/serializers/base_types.hpp"
+#include "ledger/chain/block.hpp"
+
+namespace fetch {
+namespace ledger {
+
+/**
+ * Packet format used to convey heaviest chain for node sync.
+ */
+template <class B>
+struct TimeTravelogue
+{
+  using BlockType = B;
+  using Blocks    = std::vector<BlockType>;
+
+  Blocks  blocks;
+  int64_t next_direction;
```

Comment:
> It is not clear why next direction is 64 bits long here?

No idea, borrowed the  type from the original implementation where 64-bit uint was used to convey numbers of 10K at most.

In fact, next_dir is better defined as a separate enum.

> Templating on Block here seems unneccessary since it is only ever templated on a single type?

In fact,
it is `TimeTravelogue<BlockPtr>` in `MainChain` and `TimeTravelogue<Block>` in Protocol. Can be made the same type (need to check how `shared_ptr`s are deserialized).

Reply:
Sync being only done forward, this is no longer a field to the struct.